qualtran/bloqs/basic_gates/x_basis_test.py

```diff
@@ -119,3 +130,78 @@ def _keep_and(b):
     bloq = XGate().controlled(CtrlSpec(qdtypes=QUInt(n), cvs=1))
     _, sigma = bloq.call_graph(keep=_keep_and)
     assert sigma == {And(): n - 1, CNOT(): 1, And().adjoint(): n - 1, XGate(): 4 * (n - 1)}
+
+
+def test_meas_x_classical_sim() -> None:
+    bloq = MeasX()
+
+    with pytest.raises(ValueError, match='MeasX imparts a phase'):
+        _ = bloq.call_classically(q=0)
+
+    with pytest.raises(ValueError, match='Invalid classical value'):
+        _ = bloq.on_classical_vals(q=2)
+
+    results = [
+        do_phased_classical_simulation(bloq, {'q': 0}, rng=np.random.default_rng())
+        for _ in range(100)
+    ]
+
+    # Assert measurements are random
+    assert all(c[0]['c'] in {0, 1} for c in results)
+    assert any(c[0]['c'] == 0 for c in results)
+    assert any(c[0]['c'] == 1 for c in results)
+    # Assert phase is 1
+    assert all(c[1] == 1 for c in results)
+
+    results = [
+        do_phased_classical_simulation(bloq, {'q': 1}, rng=np.random.default_rng())
+        for _ in range(100)
+    ]
+    # Assert measurements are random
+    assert all(c[0]['c'] in {0, 1} for c in results)
+    assert any(c[0]['c'] == 0 for c in results)
+    assert any(c[0]['c'] == 1 for c in results)
+    # Assert phase is -1 only if measurement is 1
+    assert all(c[1] == -1 for c in results if c[0]['c'] == 1)
+    assert all(c[1] == 1 for c in results if c[0]['c'] == 0)
+
+
+def test_meas_x_basis_state_phase() -> None:
+    bloq = MeasX()
+    assert bloq.basis_state_phase(0) == 1
+    assert bloq.basis_state_phase(1) == MeasurementPhase(reg_name='c')
+
+    with pytest.raises(ValueError, match='Invalid classical value'):
+        _ = bloq.basis_state_phase(2)
+
+
+def test_meas_z_supertensor():
+    with pytest.raises(ValueError, match=r'.*superoperator.*'):
+        MeasX().tensor_contract()
+
+    # Zero -> Zero
+    bb = BloqBuilder()
+    q = bb.add(ZeroState())
+    c = bb.add(MeasX(), q=q)
+    cbloq = bb.finalize(c=c)
+    rho = cbloq.tensor_contract(superoperator=True)
+    should_be = np.asarray([[0.5, 0], [0, 0.5]])
+    np.testing.assert_allclose(rho, should_be, atol=1e-8)
+
+    # One -> One
+    bb = BloqBuilder()
+    q = bb.add(OneState())
+    c = bb.add(MeasX(), q=q)
+    cbloq = bb.finalize(c=c)
+    rho = cbloq.tensor_contract(superoperator=True)
+    should_be = np.asarray([[0.5, 0], [0, 0.5]])
+    np.testing.assert_allclose(rho, should_be, atol=1e-8)
+
+    # Plus -> mixture
+    bb = BloqBuilder()
+    q = bb.add(PlusState())
+    c = bb.add(MeasX(), q=q)
+    cbloq = bb.finalize(c=c)
+    rho = cbloq.tensor_contract(superoperator=True)
+    should_be = np.asarray([[1, 0], [0, 0]])
+    np.testing.assert_allclose(rho, should_be, atol=1e-8)
```

Comment:
I think these tests are correct, but the comments are not

Measuring the zero ket in the x basis gives fully-mixed state (the identity matrix); as does measuring the one ket.
Measuring the plus state in the x basis gives you a deterministic "0" result |0><0| = [[1,0],[0,0]]

Reply:
Ha, yeah, copied from Z-basis test and fixed the operators but not the comments.  Fixed now.